docs/developer/variable_implementation.rst

```diff
@@ -58,3 +48,9 @@ To support operations without requiring an abundance of overloads, read-only arg
 
 - ``VariableView`` inherits ``VariableConstView``.
 - ``Variable`` is implicitly convertible to ``VariableConstView``.
+
+Binned variables
+----------------
+
+Binned variables are implemented using a specialization of ``DataModel<T>``.
```

Comment:
It might be good to at least point to the specialisations if they are not documented further.